pkg/bpmn_engine/engine_jobs.go

```diff
@@ -76,28 +111,80 @@ func findOrCreateJob(jobs []*job, id string, instance *ProcessInstanceInfo, gene
 		State:              activity.Active,
 		CreatedAt:          time.Now(),
 	}
+
 	jobs = append(jobs, &job)
+
 	return &job
 }
 
+// GetCreatedAt implements ActivatedJob
+func (aj *activatedJob) GetCreatedAt() time.Time {
+	return aj.CreatedAt
+}
+
+// GetInstanceKey implements ActivatedJob
+func (aj *activatedJob) GetInstanceKey() int64 {
+	return aj.processInstanceInfo.GetInstanceKey()
+}
+
+// GetProcessInfo implements ActivatedJob
+func (aj *activatedJob) GetProcessInfo() *ProcessInfo {
+	return aj.processInstanceInfo.GetProcessInfo()
+}
+
+// GetState implements ActivatedJob
+func (aj *activatedJob) GetState() process_instance.State {
+	return aj.processInstanceInfo.GetState()
+}
+
+// GetElementID implements ActivatedJob
+func (aj *activatedJob) GetElementId() string {
+	return aj.ElementId
+}
+
+// GetKey implements ActivatedJob
+func (aj *activatedJob) GetKey() int64 {
+	return aj.Key
+}
+
+// GetBpmnProcessID implements ActivatedJob
+func (aj *activatedJob) GetBpmnProcessId() string {
+	return aj.BpmnProcessId
+}
+
+// GetProcessDefinitionKey implements ActivatedJob
+func (aj *activatedJob) GetProcessDefinitionKey() int64 {
+	return aj.ProcessDefinitionKey
+}
+
+// GetProcessDefinitionVersion implements ActivatedJob
+func (aj *activatedJob) GetProcessDefinitionVersion() int32 {
+	return aj.ProcessDefinitionVersion
+}
+
+// GetProcessInstanceKey implements ActivatedJob
+func (aj *activatedJob) GetProcessInstanceKey() int64 {
+	return aj.ProcessInstanceKey
+}
+
 // GetVariable from the process instance's variable context
-func (activatedJob ActivatedJob) GetVariable(key string) interface{} {
-	return activatedJob.processInstanceInfo.GetVariable(key)
+func (aj *activatedJob) GetVariable(key string) interface{} {
+	return aj.processInstanceInfo.GetVariable(key)
 }
 
 // SetVariable to the process instance's variable context
-func (activatedJob ActivatedJob) SetVariable(key string, value interface{}) {
-	activatedJob.processInstanceInfo.SetVariable(key, value)
+func (aj *activatedJob) SetVariable(key string, value interface{}) {
+	aj.processInstanceInfo.SetVariable(key, value)
 }
 
 // Fail does set the state the worker missed completing the job
 // Fail and Complete mutual exclude each other
-func (activatedJob ActivatedJob) Fail(reason string) {
-	activatedJob.failHandler(reason)
+func (aj *activatedJob) Fail(reason string) {
+	aj.failHandler(reason)
 }
 
 // Complete does set the state the worker successfully completing the job
```

Comment:
The documentation for GetVariable(), SetVariable(), Complete() and Fail() are more comprehensive here,
then in the interface declaration. would you please move them to the interface, also to make the implementation methods consistently documented like this "... implements ActivatedJob"

Reply:
Could you please re-review? and see if that's what you expected